universal-application-tool-0.0.1/app/forms/TextQuestionForm.java

```diff
@@ -0,0 +1,58 @@
+package forms;
+
+import java.util.OptionalInt;
+import services.question.InvalidQuestionTypeException;
+import services.question.QuestionDefinitionBuilder;
+import services.question.TextQuestionDefinition;
+
+public class TextQuestionForm extends QuestionForm {
+  private OptionalInt textMinLength;
+  private OptionalInt textMaxLength;
+
+  public TextQuestionForm() {
+    super();
+    textMinLength = OptionalInt.empty();
+    textMaxLength = OptionalInt.empty();
+    // TODO(https://github.com/seattle-uat/civiform/issues/590): Use QuestionType instead of String
+    //  for this?
+    setQuestionType("TEXT");
+  }
+
+  public TextQuestionForm(TextQuestionDefinition qd) {
+    super(qd);
+    textMinLength = qd.getMinLength();
+    textMaxLength = qd.getMaxLength();
+  }
+
+  public OptionalInt getTextMinLength() {
+    return textMinLength;
+  }
+
+  public void setTextMinLength(int textMinLength) {
```

Comment:
Questions (non-blocking): I am not super familiar with html form so this question is just for my understanding. What are the types supported in the html form? I originally thought only text is supported so arguments here needs to be string as well.

Reply:
You might actually be right... I thought that the Play Framework FormFactory stuff might give us some parsing out of the box (since you are right that the request just gets sent over as strings, as far as I understand).

Let me do some more research.

Reply:
I think Play may actually do the conversion for us. My only concern there is that if the parsing fails, Play likely throws a runtime exception that results in a 5XX server error. But we'd like to handle that as expected user input errors and show an error message on the same page.

Reply:
That is a good point!

So what do you think - safer to accept a string and manually parse? Then re-render with an error message?

Reply:
It _might_ be overkill, since in the admin questions form, we can (and do) use an input field with `type="number" step="1"`, which makes it essentially impossible to enter a non-int. But it is still probably safer to check it ourselves.

Reply:
Ah I see. Yeah, personally I'd manually parse it, but don't feel strongly.

Reply:
It's not clear where an error thrown in `setMinLength` would surface - I'm guessing it would happen in the `form.bindFromRequest` call?

I'm going to leave it as is for now since I think it'd be a non-trivial effort to pipe it up correctly.